lib/galaxy/managers/library_datasets.py

```diff
@@ -0,0 +1,234 @@
+"""Manager and Serializer for library datasets."""
+import logging
+
+from galaxy import util
+
+from galaxy.exceptions import InternalServerError
+from galaxy.exceptions import ItemAccessibilityException
+from galaxy.exceptions import InsufficientPermissionsException
+from galaxy.exceptions import ObjectNotFound
+from galaxy.exceptions import RequestParameterInvalidException
+from galaxy.managers import tags
+from galaxy.util import validation
+
+log = logging.getLogger(__name__)
+
+
+class LibraryDatasetsManager(object):
+    """Interface/service object for interacting with library datasets."""
+
+    def __init__(self, app):
+        self.app = app
+        self.tag_manager = tags.GalaxyTagManager(app.model.context)
+
+    def get(self, trans, decoded_library_dataset_id, check_accessible=True):
+        """
+        Get the library dataset from the DB.
+
+        :param  decoded_library_dataset_id: decoded library dataset id
+        :type   decoded_library_dataset_id: int
+        :param  check_accessible:           flag whether to check that user can access item
+        :type   check_accessible:           bool
+
+        :returns:   the requested library dataset
+        :rtype:     galaxy.model.LibraryDataset
+        """
+        try:
+            ld = trans.sa_session.query(trans.app.model.LibraryDataset).filter(trans.app.model.LibraryDataset.table.c.id == decoded_library_dataset_id).one()
+        except Exception as e:
+            raise InternalServerError('Error loading from the database.' + str(e))
+        ld = self.secure(trans, ld, check_accessible)
+        return ld
+
+    def update(self, trans, ld, payload):
+        """
+        Update the given library dataset - the latest linked ldda.
+        Updating older lddas (versions) is not supported.
+
+        :param  ld: library dataset to change
+        :type   ld: LibraryDataset
+        :param  check_ownership:           flag whether to check that user owns the item
+        :type   check_ownership:           bool
+        :param  check_accessible:          flag whether to check that user can access item
+        :type   check_accessible:          bool
```

Comment:
`check_ownership` and `check_accessible` are not parameters of this function, documentation for `payload` is missing instead.